Mathlib/Analysis/NormedSpace/HahnBanach/SeparatingDual.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2023 Sébastien Gouëzel All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.NormedSpace.HahnBanach.Extension
+import Mathlib.Analysis.NormedSpace.HahnBanach.Separation
+
+/-!
+# Spaces with separating dual
+
+We introduce a typeclass `SeparatingDual R V`, registering that the points of the topological
+module `V` over `R` can be separated by continuous linear forms.
+
+This property is satisfied for normed spaces over `ℝ` or `ℂ` (by the analytic Hahn-Banach theorem)
+and for locally convex topological spaces over `ℝ` (by the geometric Hahn-Banach theorem).
+
+Under the assumption `SeparatingDual R V`, we show in
+`SeparatingDual.exists_continuousLinearMap_apply_eq` that the group of continuous linear
+equivalences acts transitively on the set of nonzero vectors.
+-/
+
+/-- When `E` is a topological module over a topological ring `R`, the class `SeparatingDual R E`
+registers that continuous linear forms on `E` separate points of `E`. -/
+class SeparatingDual (R : Type _) (V : Type _) [Ring R] [AddCommGroup V] [TopologicalSpace V]
+  [TopologicalSpace R] [Module R V] : Prop :=
+  /-- Any nonzero vector can be mapped by a continuous linear map to a nonzero scalar. -/
+  exists_ne_zero' : ∀ (x : V), x ≠ 0 → ∃ f : V →L[R] R, f x ≠ 0
```

Comment:
Should we generalize the definition of `NormedSpace.dual` so that it can be used here?

Reply:
I'm not sure I see the point of `NormedSpace.dual` (it's a non-reducible type synonym, but with the same typeclasses as the original one). In doubt, I'd rather stay with the basic version.